### PR TITLE
Install scylla-operator components using its Helm charts

### PR DESCRIPTION
**Pull request by vponomaryov**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/cluster_k8s/__init__.py [hunk 1]
```diff
@@ -149,6 +160,29 @@ def deploy_cert_manager(self) -> None:
...
+        if chart_version in ("", "latest"):
+            latest_version_raw = self.helm(
+                "search repo scylla-operator/scylla-operator --devel -o yaml")
+            latest_version = yaml.safe_load(latest_version_raw)
+            assert isinstance(
+                latest_version, list), f"Expected list of data, got: {type(latest_version)}"
+            assert len(latest_version) == 1, "Expected only one element in the list of versions"
+            assert "version" in latest_version[0], "Expected presence of 'version' key"
+            chart_version = latest_version[0]["version"].strip()
+            LOGGER.debug(f"Using automatically found following "
```
Comment (bentsi) [inline sdcm/cluster_k8s/__init__.py:184]: change to info log level

File: sdcm/cluster_k8s/__init__.py [hunk 2]
```diff
@@ -149,6 +160,29 @@ def deploy_cert_manager(self) -> None:
...
+                         f"latest scylla operator chart version: {chart_version}")
+        else:
+            LOGGER.debug(f"Using following predefined scylla operator "
```
Comment (bentsi) [inline sdcm/cluster_k8s/__init__.py:187]: change to info log level

File: sdcm/sct_config.py [hunk 3]
```diff
@@ -642,6 +642,11 @@ class SCTConfiguration(dict):
         dict(name="k8s_scylla_operator_docker_image", env="SCT_K8S_SCYLLA_OPERATOR_DOCKER_IMAGE", type=str,
              help=""),
 
+        dict(name="k8s_scylla_operator_helm_repo", env="SCT_K8S_SCYLLA_OPERATOR_HELM_REPO", type=str,
+             help=""),
```
Comment (bentsi) [inline sdcm/sct_config.py:646]: add help message

File: sdcm/sct_config.py [hunk 4]
```diff
@@ -642,6 +642,11 @@ class SCTConfiguration(dict):
...
+        dict(name="k8s_scylla_operator_chart_version", env="SCT_K8S_SCYLLA_OPERATOR_CHART_VERSION", type=str,
+             help=""),
```
Comment (bentsi) [inline sdcm/sct_config.py:648]: add help message

File: sdcm/utils/k8s.py [hunk 5]
```diff
@@ -316,18 +316,64 @@ def helm_container_run_args(self) -> dict:
 
     @cached_property
     def _helm_container(self) -> Container:
-        return ContainerManager.run_container(self, "helm")
+        c = ContainerManager.run_container(self, "helm")
```
Comment (bentsi) [inline sdcm/utils/k8s.py:319]: `c` -> `container`

File: sdcm/cluster_k8s/__init__.py [hunk 6]
```diff
@@ -185,10 +188,55 @@ def _scylla_operator_chart_version(self):
 
     @log_run_info
     def deploy_scylla_manager(self) -> None:
+        # Calculate options values which must be set
+        #
+        # image.tag                  -> self.params.get('mgmt_docker_image').split(':')[-1]
+        # scylla.scyllaImage.tag     -> self.params.get('scylla_version')
+        # scylla.agentImage.tag      -> self.params.get('scylla_mgmt_agent_version')
```
Comment (bentsi) [inline sdcm/cluster_k8s/__init__.py:197]: let's use the default Scylla version as in "standard" manager installation

File: sdcm/cluster_k8s/gke.py [hunk 7]
```diff
@@ -34,7 +34,7 @@
 GKE_API_CALL_QUEUE_SIZE = 1000  # ops
 GKE_URLLIB_RETRY = 6  # How many times api request is retried before reporting failure
 
-SCYLLA_CLUSTER_CONFIG = sct_abs_path("sdcm/k8s_configs/cluster-gke.yaml")
+SCYLLA_CLUSTER_CONFIG = "/apps/k8s_configs/gke-cluster-chart-values.yaml"
```
Comment (bentsi) [inline sdcm/cluster_k8s/gke.py:37]: let's move `/apps/k8s_configs/` to a global var

**Review by bentsi — CHANGES_REQUESTED** [review]
Well done! please address minor comments

File: sdcm/cluster_k8s/__init__.py (hunk 1 above)
Comment (vponomaryov) [reply sdcm/cluster_k8s/__init__.py:184]: Done

File: sdcm/cluster_k8s/__init__.py (hunk 2 above)
Comment (vponomaryov) [reply sdcm/cluster_k8s/__init__.py:187]: Done

File: sdcm/utils/k8s.py (hunk 5 above)
Comment (vponomaryov) [reply sdcm/utils/k8s.py:319]: Done

File: sdcm/sct_config.py (hunk 3 above)
Comment (vponomaryov) [reply sdcm/sct_config.py:646]: Done

File: sdcm/sct_config.py (hunk 4 above)
Comment (vponomaryov) [reply sdcm/sct_config.py:648]: Done

File: sdcm/cluster_k8s/__init__.py (hunk 6 above)
Comment (vponomaryov) [reply sdcm/cluster_k8s/__init__.py:197]: done

File: sdcm/cluster_k8s/gke.py (hunk 7 above)
Comment (vponomaryov) [reply sdcm/cluster_k8s/gke.py]: Done

**Review by bentsi — APPROVED** [review]

**Pull request merged**